modules/cover-image-workflowoperation/src/main/java/org/opencastproject/workflow/handler/coverimage/CoverImageWorkflowOperationHandlerBase.java

```diff
@@ -302,57 +304,87 @@ protected String loadXsl(WorkflowOperationInstance operation) throws WorkflowOpe
     }
   }
 
-  protected String getMetadataXml(MediaPackage mp) {
-    StaticMetadata metadata = getStaticMetadataService().getMetadata(mp);
+  protected String getMetadataXml(MediaPackage mp, WorkflowOperationInstance operation) {
+    //get specified episode/series flavor
+    String configuredEpisodeFlavor = StringUtils.trimToEmpty(operation.getConfiguration(EPISODE_FLAVOR));
+    String configuredSeriesFlavor = StringUtils.trimToEmpty(operation.getConfiguration(SERIES_FLAVOR));
 
+    //use dublincore/episode and dublincore/series metadata-catalogs as default
+    if (("").equals(configuredEpisodeFlavor)) {
+      configuredEpisodeFlavor = "dublincore/episode";
+    }
+    if (("").equals(configuredSeriesFlavor)) {
+      configuredSeriesFlavor = "dublincore/series";
+    }
+
+    String episodeFlavorType = configuredEpisodeFlavor.split("/")[0];
+    String episodeFlavorSubtype = configuredEpisodeFlavor.split("/")[1];
+    String seriesFlavorType = configuredSeriesFlavor.split("/")[0];
+    String seriesFlavorSubtype = configuredSeriesFlavor.split("/")[1];
+
+    //Get episode metadata-catalog
+    Catalog[] catalogs = mp.
+            getCatalogs(new MediaPackageElementFlavor(episodeFlavorType, StringUtils.lowerCase(episodeFlavorSubtype)));
+
+   //load metadata-catalog
+    DublinCoreCatalog dc = DublinCoreUtil.loadDublinCore(getWorkspace(), catalogs[0]);
+    Map<EName, List<DublinCoreValue>> data = dc.getValues();
+
+    //build xml from metadata
     StringBuilder xml = new StringBuilder();
     xml.append("<metadata xmlns:dcterms=\"http://purl.org/dc/terms/\">");
 
-    for (String title : getFirstMetadataValue(metadata.getTitles()))
-      appendXml(xml, "title", title);
-    for (String description : getFirstMetadataValue(metadata.getDescription()))
-      appendXml(xml, "description", description);
-    for (String language : metadata.getLanguage())
-      appendXml(xml, "language", language);
-    for (Date created : metadata.getCreated())
-      /* Method formatDate of org.apache.xalan.lib.ExsltDatetime requires the format CCYY-MM-DDThh:mm:ss */
-      appendXml(xml, "date", new SimpleDateFormat("YYYY-MM-dd'T'HH:mm:ss").format(created));
-    for (Date[] period : metadata.getTemporalPeriod()) {
-      if (period[0] != null) {
-        appendXml(xml, "start", new SimpleDateFormat("YYYY-MM-dd'T'HH:mm:ss").format(period[0]));
-      }
-      if (period[1] != null) {
-        appendXml(xml, "end", new SimpleDateFormat("YYYY-MM-dd'T'HH:mm:ss").format(period[1]));
+    for (Map.Entry<EName, List<DublinCoreValue>> entry : data.entrySet()) {
+      String currentKey = entry.getKey().getLocalName();
+      switch(currentKey) {
+        case "creator": appendXml(xml, "creators", entry.getValue().get(0).getValue()); break;
```

Comment:
All these fields can exist not at all or multiple times. Will this work if they do not exist? Onbly using the first is probably okay but should be documented.

Reply:
Also the order of the list isn't fixed. I guess, we have to provide all values as list.

Reply:
@lkiesow If they do not exist, they are not beeing added to the xml, so it should be fine. There were no issues during my tests with non existing fields. I also added a note in the documentation.

@wsmirnow You mean the order of the fields in the resulting xml should stay intact?

Reply:
iirc, the order isn't guaranteed in an XML document anyway.

Reply:
I would prefer the same order. It may be technically not possible. If you put only the first value of the list, you may end up with different values each call of the handler.

Reply:
Now I'm confused as well :D
The code should be deterministic, shouldn't it?
So you would get the same. You just don't get all of them.

@jduehring, what happens if you iterate e.g. over multiple creator fields and call `appendXml` every time?

Reply:
Not sure how to test this. What would be the easiest way to get an episode containing metadata with multiple creator fields? Should this happen when I add multiple creators via the admin-ui? So for each creator one creator field?
Currently there seems to be a bug so that this does not happen (#1224).
In theory my code should currently add  the same field multiple times, which is also not ideal, i guess.

Reply:
Sorry for the late reply :see_no_evil: 

The following cURL command will ingest media with two titles (which will mostly be ignored by Opencast code and just the first is usually being used) and multiple creators (which Opencast can handle just fine… mostly):

```sh
curl -i -u admin:opencast \
  https://develop.opencast.org/ingest/addMediaPackage \
  -F 'flavor=presentation/source' \
  -F 'BODY=@sintel_trailer-1080p.mp4' \
  -F title="Title 1" \
  -F title="Title 2" \
  -F creator="Creator 1" \
  -F creator="Creator 2"
```